src/Core/Impl/IO/DirectoryInfoProxy.cs

```diff
@@ -40,6 +42,17 @@ public sealed class DirectoryInfoProxy : IDirectoryInfo {
                 .EnumerateFileSystemInfos()
                 .Select(CreateFileSystemInfoProxy);
 
+        public IEnumerable<IFileSystemInfo> EnumerateFileSystemInfos(string[] includeFiles, string[] excludeFiles) {
+            Matcher matcher = new Matcher();
+            matcher.AddIncludePatterns(includeFiles.IsNullOrEmpty() ? new[] { "**/*" } : includeFiles);
+            matcher.AddExcludePatterns(excludeFiles ?? Enumerable.Empty<string>());
+            var matchResult = matcher.Execute(new DirectoryInfoWrapper(_directoryInfo));
```

Comment:
Do we care about this case: https://github.com/aspnet/Extensions/issues/449 ?

Reply:
It should be signaled back to the user. 
Still, I don't think it's frequent for users to have sub directories inaccessible inside the workspace

Reply:
Typically this exception is thrown if some other tool or an anti-malware scanner is running. Consider user running PyLint from VSCode while you are indexing. So I'd add a retry at least and definitely we don't want to crash LS just b/c some file is not accessible.